### PR TITLE
Migrate Wiki from https://developer.jboss.org/docs/DOC-11431

### LINKED ISSUE #61: Migrate old JBoss Wiki into this repo
The wiki is here:

https://developer.jboss.org/docs/DOC-11431

It's not even editable any more... so this looks like a good place.

### PR DESCRIPTION
**Pull request by jajik**
close #61 

This PR only migrates the wiki without any updates to the content (except some minor formatting tweaks and typo fixes). We should eventually go through the content[^1] and also rework the structure of the whole documentation.

[^1]: There are also some urls that are out of date. I let them as-is until we figure out what with them.

### CODE REVIEW

File: docs/src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -53,3 +53,8 @@ include::faq.adoc[]
 
 include::migration.adoc[]
 
+= Wiki
+
+For more techical and detailed information you can look at our
+xref:wiki/wiki.adoc[Wiki].
```
Comment (rhusar) [inline docs/src/main/asciidoc/index.adoc:59]: Let's rename the wiki.adoc to index.adoc so it will be just https://docs.modcluster.io/something/

**Review by rhusar — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/index.adoc (hunk 1 above)
Comment (jajik) [reply docs/src/main/asciidoc/index.adoc]: Done. The only remaining "wiki" mentions are in the path. I would let it be like this for now.

**Review by rhusar — CHANGES_REQUESTED** [review]
OK - one last thing - please rename the 'wiki' directory to 'developer' so that the URL - https://docs.modcluster.io/developer - will be the developer resources.

**Review by rhusar — APPROVED** [review]
LGTM now.

**Comment by rhusar** [comment]
Big thanks @jajik !

**Pull request merged**